vectordb_bench/backend/clients/treedb/treedb.py

```
@@ -519,6 +542,47 @@ def _validate_config_shape(self) -> None:
         msg = f"TreeDB vector-index benchmark route does not support query_mode={mode!r}"
         raise ValueError(msg)
 
+    def _transport(self) -> str:
+        return getattr(self, "transport", "document_service")
+
+    def _partition_request(self, path: str, payload: dict | None = None) -> dict:
+        data = None if payload is None else json.dumps(payload, separators=(",", ":")).encode()
+        request = Request(self.base_url.rstrip("/") + path, data=data, method="GET" if data is None else "POST")
+        if data is not None:
+            request.add_header("Content-Type", "application/json")
+        with urlopen(request, timeout=self.timeout) as response:  # nosec B310 -- explicit user-selected local bridge
```

Ruff (0.16.2)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)